eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClient.cs

```diff
@@ -55,6 +66,23 @@ public BasicTypeSpecClient(Uri endpoint, AzureKeyCredential keyCredential, Basic
             Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) });
         }
 
+        /// <summary> Initializes a new instance of BasicTypeSpecClient. </summary>
+        /// <param name="endpoint"> Service endpoint. </param>
+        /// <param name="tokenCredential"> A credential used to authenticate to the service. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="tokenCredential"/> is null. </exception>
+        public BasicTypeSpecClient(Uri endpoint, TokenCredential tokenCredential, BasicTypeSpecClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(tokenCredential, nameof(tokenCredential));
+
+            options ??= new BasicTypeSpecClientOptions();
+
+            _endpoint = endpoint;
+            _tokenCredential = tokenCredential;
+            Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) });
```

Comment:
Shouldn't this be creating a bearer token policy using the tokenCredential?

Reply:
Let's please add at least two tests - one for KeyCredential and one for TokenCredential constructor bodies.

Reply:
!!! this means the implementation in MGC is incorrect.
Let me check and add test cases

Reply:
This has been fixed and test cases have been added